src/App.tsx

```diff
@@ -1,14 +1,55 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Status } from './types/Status';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+export const filteredQueryInput = (
+  comparedTodos: Todo[],
+  query: string,
+): Todo[] => {
+  const visibleTodos = [...comparedTodos];
+
+  if (query) {
+    return visibleTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+
+  return visibleTodos;
+};
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  const [eyeOnClick, setEyeOnClick] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then(data => {
+      setTodos(data);
+      setIsLoading(false);
+    });
+  }, []);
+
+  let comparedTodos = todos;
+
+  if (filter === Status.Active) {
+    comparedTodos = comparedTodos.filter(todo => !todo.completed);
+  } else if (filter === Status.Completed) {
+    comparedTodos = comparedTodos.filter(todo => todo.completed);
+  }
```

Comment:
- Create a helper or extend an existing one and use it.
- Use switch case instead of if/else